src/components/Pagination/Pagination.tsx

```diff
@@ -1 +1,99 @@
-export const Pagination = () => {};
+import { getNumbers } from '../../utils';
+
+type Props = {
+  total: number,
+  perPage: number,
+  currentPage: number,
+  onPageChange: (tabId: number) => void;
+};
+
+export const Pagination: React.FC<Props> = ({
+  total,
+  perPage,
+  currentPage,
+  onPageChange,
+}) => {
+  const lastPage = Math.ceil(total / perPage);
+  const pagesArray = getNumbers(1, lastPage);
+  const itemsArray = getNumbers(1, total);
+
+  const itemSeparationForPages = itemsArray.filter(item => {
+    if (currentPage === 1) {
```

Comment:
This filtering logic can be replaced with a simpler implementation. Calculate the start and end index and use slice method.